### PR TITLE
iolog: allow to read_iolog from unix socket

### PR DESCRIPTION
**Pull request by aclamk**
Checks is file provided via --read_iolog parameter is a unix socket.
	In such case connect to it and fetch data from there.

Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

**Comment by sitsofe** [comment]
I'm afraid you've broken the Windows build: https://ci.appveyor.com/project/axboe/fio/build/1.0.739/job/v9x00qtk508epaw1 ...

File: iolog.c [hunk 1]
```diff
@@ -485,16 +492,48 @@ static bool read_iolog2(struct thread_data *td, FILE *f)
 	return true;
 }
 
+#ifndef S_ISSOCK
```
Comment (axboe) [inline iolog.c]: This should probably go in os/windows/posix/include/sys/stat.h

File: iolog.c [hunk 2]
```diff
@@ -485,16 +492,48 @@ static bool read_iolog2(struct thread_data *td, FILE *f)
...
+#define S_ISSOCK(x) 0
+#endif
+
+static bool is_socket(const char* path) {
```
Comment (axboe) [inline iolog.c]: Fio style is to place the * next the variable, and the opening brace on the next line (except for if). This should be:

```static bool is_socket(const char *path)
{
[...]
```

and ditto for other locations.

File: iolog.c [hunk 3]
```diff
@@ -485,16 +492,48 @@ static bool read_iolog2(struct thread_data *td, FILE *f)
...
+}
+
+static int open_socket(const char* path) {
+	int fd = socket(AF_UNIX, SOCK_STREAM, 0);
+	struct sockaddr_un addr;
+	if (fd < 0)
+		return fd;
+	addr.sun_family = AF_UNIX;
+	strncpy(addr.sun_path, path, sizeof(addr.sun_path));
+	if (0 == connect(fd, (const struct sockaddr *)&addr, strlen(path) + sizeof(addr.sun_family)))
```
Comment (axboe) [inline iolog.c]: No reverse checking please. Make that connect() == 0 or !connect().

File: iolog.c [hunk 4]
```diff
@@ -485,16 +492,48 @@ static bool read_iolog2(struct thread_data *td, FILE *f)
...
+}
+
 /*
  * open iolog, check version, and call appropriate parser
  */
 static bool init_iolog_read(struct thread_data *td)
 {
 	char buffer[256], *p;
-	FILE *f;
+	FILE *f = 0;
```
Comment (axboe) [inline iolog.c]: Pointers are NULL.

**Comment by sitsofe** [comment]
@aclamk could you squash these into one?

**Comment by axboe** [comment]
That would be ideal (squashing), and you could even make the S_ISSOCK change a separate pre-patch.

**Comment by aclamk** [comment]
https://github.com/axboe/fio/pull/637 adds S_ISSOCK macro.

**Comment by axboe** [comment]
Pulled, thanks.

**Pull request merged**